src/openrct2/entity/Peep.cpp

```diff
@@ -1608,7 +1608,11 @@ void Peep::FormatNameTo(Formatter& ft) const
 
         if ((!isStaff && showGuestNames) || (isStaff && showStaffNames))
         {
-            auto realNameStringId = GetRealNameStringIDFromPeepID(PeepId);
+            auto nameId = PeepId;
+            if (isStaff)
+                nameId *= 256 * EnumValue(staff->AssignedStaffType) + 1;
```

Comment:
Maybe its better to use a prime to multiply so there is less of a pattern.

Reply:
What kind of prime are you thinking? Close primes are 251, 257, 263, but I don't think that would make too much of a difference with how this is currently set up.

Reply:
I'm not sure it makes a difference, either. If [this](https://youtu.be/12EqP3VL8FY?t=79) is still accurate and there are 16,384 total names, then to get the most spread over available names for staff, you'd want to offset each staff type by 4,096.